### PR TITLE
Remove Obsolete BwC Logic from BlobStoreRepository

### PR DESCRIPTION
**Pull request by original-brownbear**
* We can't restore 1.3.3 files anyway -> no point in doing the dance of computing a hash here
* Some other minor+obvious cleanups

### CODE REVIEW

**Comment by original-brownbear** [comment]
Jenkins run elasticsearch-ci/1

File: server/src/main/java/org/elasticsearch/repositories/blobstore/FileRestoreContext.java [hunk 1]
```diff
@@ -118,7 +117,7 @@ public void restore(SnapshotFiles snapshotFiles) throws IOException {
                 // store can still have existing files but they will be deleted just before being
                 // restored.
                 recoveryTargetMetadata = indexShard.snapshotStoreMetadata();
-            } catch (org.apache.lucene.index.IndexNotFoundException e) {
+            } catch (IndexNotFoundException e) {
```
Comment (ywelsch) [inline server/src/main/java/org/elasticsearch/repositories/blobstore/FileRestoreContext.java]: I wanted to have this explicitly as a fully qualified name, in order to ensure that it does not fall victim to a refactoring to Elasticsearch's IndexNotFoundException, which recently happened.

Please revert this.

**Review by ywelsch — CHANGES_REQUESTED** [review]

File: server/src/main/java/org/elasticsearch/repositories/blobstore/FileRestoreContext.java (hunk 1 above)
Comment (original-brownbear) [reply server/src/main/java/org/elasticsearch/repositories/blobstore/FileRestoreContext.java]: reverted -> https://github.com/elastic/elasticsearch/pull/42193/commits/f2ab652fb1a49e8dc926436dcf4f79a9a176ffb2

**Review by ywelsch — APPROVED** [review]
LGTM

**Comment by original-brownbear** [comment]
Jenkins run elasticsearch-ci/1

**Comment by original-brownbear** [comment]
thanks @ywelsch !

**Pull request merged**